pennylane/data/__init__.py

```diff
@@ -133,6 +133,9 @@
 >>> my_dataset = Dataset.open("~/dataset/my_dataset/h5", mode="copy")
 >>> my_dataset.new_attribute = "abc"
 
+Since opened datasets stream data from the disk, it is not possible to simultaneously access the same dataset
+from multiple variables, from separately running scripts, or multiple notebooks. To get around this, either make a
+copy of the dataset in the disk or access the dataset using ``mode="copy"``.
```

Comment:
Is this mode just for reading, or does it allow for writing as well? Could be made clearer. Does this also work for Windows?

Reply:
It's just for reading to memory. It is possible to edit while in memory and then write to a new dataset or other dataset opened with with `mode='w'`

Windows support: I haven't tested this, but I don't see why it wouldn't be possible on Windows :thinking: did you have something in mind?

Reply:
I haven't looked at how `copy` mode works internally, but it might be nice to test it on Windows to ensure there's no permission mismatch!

Reply:
Yup, had @Chiffafox check it on Windows and there don't seem to be any issues. :+1: